### PR TITLE
split `requirements-dev.txt` into first and third party dev dependencies

### PR DESCRIPTION
**Pull request by zacharyburnett**
This PR splits the dev dependencies into `requirements-dev.txt` and `requirements-dev-thirdparty.txt`

**Checklist**
- [ ] ~added entry in `CHANGES.rst` under the corresponding subsection~
- [ ] ~updated relevant tests~
- [ ] ~updated relevant documentation~
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

**Comment by zacharyburnett** [comment]
wait until #843

File: tox.ini [hunk 1]
```diff
@@ -64,7 +65,8 @@ deps =
 commands_pre =
     oldestdeps: minimum_dependencies romancal --filename requirements-min.txt
     oldestdeps: pip install -r requirements-min.txt
-    devdeps: pip install -r requirements-dev.txt -U --upgrade-strategy eager
+    stdevdeps: pip install -r requirements-dev-st.txt -U --upgrade-strategy eager
+    devdeps: pip install -r requirements-dev-thirdparty.txt -U --upgrade-strategy eager
```
Comment (WilliamJamieson) [inline tox.ini:69]: This should include the `requirements-dev-st.txt` in addition to the third party ones.

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 1 above)
Comment (zacharyburnett) [inline tox.ini:69]:
```suggestion
    devdeps: pip install -r requirements-dev-thirdparty.txt -r requirements-dev-st.txt -U --upgrade-strategy eager
```

**Review by schlafly — APPROVED** [review]
Looks good to me, though I'm mostly deferring to the experts!

**Comment by schlafly** [comment]
@zacharyburnett , can we add a changelog entry and then go ahead and merge this?

**Comment by ddavis-stsci** [comment]
> @zacharyburnett , can we add a changelog entry and then go ahead and merge this?

The change log is fine but we also need and update to the README.md to explain to the users how to use the new files when installing the software.

File: README.md [hunk 2]
```diff
@@ -102,6 +102,12 @@ tools used for linting and reproducible environments run:
 
 This will install `tox` and the `pre-commit` tools in addition to all development dependencies.
 
+Development versions of dependencies are listed in `requirements-dev-st.txt` and `requirements-dev-thirdparty.txt`, containing STScI and third-party packages respectively.
+To install these dependencies, run the following:
+
+    pip install -r requirements-dev-st.txt
+    pip install -r requirements-dev-thirdparty.txt
```
Comment (zacharyburnett) [inline README.md:109]: how is this?

**Review by ddavis-stsci — APPROVED** [review]
LGTM

**Pull request merged**